File: zig/src/indicators/core/outputs/shape.zig
/// Identifies the data shape of an indicator output.
pub const Shape = enum(u8) {
    /// Holds a time stamp and a value.
    scalar = 1,
    /// Holds a time stamp and two values representing upper and lower lines of a band.
    band = 2,
    /// Holds a time stamp and an array of values representing a heat-map column.
    heatmap = 3,
    /// Holds a time stamp and an ordered, variable-length sequence of (offset, value) points.
    polyline = 4,
    /// Holds a time stamp and a variable-length set of (value, offset, strength) entries.
    levels = 5,

    pub fn asStr(self: Shape) []const u8 {
        return switch (self) {
            .scalar => "scalar",
            .band => "band",
            .heatmap => "heatmap",
            .polyline => "polyline",
            .levels => "levels",
        };
    }
};

File: rust/src/indicators/core/outputs/shape.rs
/// Identifies the data shape of an indicator output.
#[derive(Debug, Clone, Copy, PartialEq, Eq)]
#[repr(u8)]
pub enum Shape {
    /// Holds a time stamp and a value.
    Scalar = 1,
    /// Holds a time stamp and two values representing upper and lower lines of a band.
    Band = 2,
    /// Holds a time stamp and an array of values representing a heat-map column.
    Heatmap = 3,
    /// Holds a time stamp and an ordered, variable-length sequence of (offset, value) points.
    Polyline = 4,
}

impl Shape {
    /// Returns the string representation.
    pub fn as_str(&self) -> &'static str {
        match self {
            Shape::Scalar => "scalar",
            Shape::Band => "band",
            Shape::Heatmap => "heatmap",
            Shape::Polyline => "polyline",
        }
    }
}

impl std::fmt::Display for Shape {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        f.write_str(self.as_str())
    }
}
